### ISSUE TITLE
llvmPackages_7.compiler-rt fails to produce output path on armv7l

### ISSUE THREAD

**Issue by Atemu** [NixOS/nixpkgs#87266]
**Describe the bug**

Same issue as #66693 or #85622, but on armv7l

**Additional context**

On @thefloweringash's hydra, the clang build first started failing with this error after going from 84cf00f98031e93f389f1eb93c4a7374a33cc0a9 to 7e07846d99bed4bd7272b1626038e77a78aa36f6.  
I diffed the range and found 74f3b25896129f20f7af62fd7ccbb9fd1046761b to be the most likely culprit.

It enabled some cmakeFlags globally which used to only be enabled on darwin and useLLVM systems.  
I don't know what exactly these do but, I was able to build `llvmPackages_7.compiler-rt` on my rpi2 by disabling them on aarch32:

```diff
diff --git a/pkgs/development/compilers/llvm/7/compiler-rt.nix b/pkgs/development/compilers/llvm/7/compiler-rt.nix
index 055fb01215f..e30a5171008 100644
--- a/pkgs/development/compilers/llvm/7/compiler-rt.nix
+++ b/pkgs/development/compilers/llvm/7/compiler-rt.nix
@@ -20,7 +20,7 @@ stdenv.mkDerivation {
     "-DSCUDO_DEFAULT_OPTIONS=DeleteSizeMismatch=0:DeallocationTypeMismatch=0"
   ];
 
-  cmakeFlags = [
+  cmakeFlags = stdenv.lib.optionals (!stdenv.isAarch32) [
     "-DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON"
     "-DCMAKE_C_COMPILER_TARGET=${stdenv.hostPlatform.config}"
     "-DCMAKE_ASM_COMPILER_TARGET=${stdenv.hostPlatform.config}"
```

Since I have no idea why this works and what would be affected, I'd rather have someone look over this.

**Notify maintainers**

@TravisWhitaker @matthewbauer

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"armv7l-linux"`
 - host os: `Linux 5.4.25, NixOS, 20.09pre-git (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(root): `"nixos-unstable2, nixos-20.09pre223023.fce7562cf46, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: llvmPackages
# a list of nixos modules affected by the problem
module:
```

**Comment by matthewbauer** [comment]
we probably need to add armv7l here:

https://github.com/llvm/llvm-project/blob/b5973d20b388ab7aa7590757177454e17fb2e59b/compiler-rt/cmake/builtin-config-ix.cmake#L27

**Comment by Atemu** [comment]
I tried but it fails with this error now:

```
CMake Error at cmake/Modules/AddCompilerRT.cmake:273 (add_library):
  No SOURCES given to target: clang_rt.builtins-armv7l
Call Stack (most recent call first):
  lib/builtins/CMakeLists.txt:602 (add_compiler_rt_runtime)
```

I guess it can't find a default target for armv7l somehow?

Not really familiar with cmake unfortunately

**Comment by Atemu** [comment]
I made some progress, it seems to be taking the target arch from the triplet given to `CMAKE_C_COMPILER_TARGET`, even if you override the target arch with a flag. Printf debugging showed me that it does this halfway through the configure phase, it's armhf in the beginning if I set the flag.  

Setting the compiler target to `armhf-unknown-linux-gnueabihf` instead of `armv7l-unknown-linux-gnueabihf` makes it find its target and compile successfully.

Does  `CMAKE_C_COMPILER_TARGET` have any importance outside the build process?
